webblog/frontend/src/components/accounts/Register.js

```diff
@@ -0,0 +1,109 @@
+
+export class Register extends Component {
+    static propTypes = {
+        register: PropTypes.func.isRequired,
+        isAuthenticated: PropTypes.bool
+    };
+    state = {
+        username: "",
+        email: "",
+        password: "",
+        password2: ""
+    };
+
+    onSubmit = e => {
+        e.preventDefault();
+        const {username, email, password, password2} = this.state;
+        if (password !== password2) {
+            this.props.createMessage({passwordNotMatch: "Passwords do not match"});
+        } else {
+            const newUser = {
+                username,
+                password,
+                email
+            };
+            this.props.register(newUser);
+        }
+    };
+
+    onChange = e => this.setState({[e.target.name]: e.target.value});
+
+    render() {
+        if (this.props.isAuthenticated) {
+            return <Redirect to="/"/>;
+        }
+        const {username, email, password, password2} = this.state;
+        return (
+            <div className="col-md-6 m-auto">
+                <div className="card card-body mt-5">
+                    <h2 className="text-center">Register</h2>
+                    <form onSubmit={this.onSubmit}>
+                        <div className="form-group">
+                            <label>Username</label>
+                            <input
+                                type="text"
+                                className="form-control"
+                                name="username"
+                                onChange={this.onChange}
+                                value={username}
+                            />
+                        </div>
+                        <div className="form-group">
+                            <label>Email</label>
+                            <input
+                                type="email"
+                                className="form-control"
+                                name="email"
+                                onChange={this.onChange}
+                                value={email}
+                            />
+                        </div>
+                        <div className="form-group">
+                            <label>Password</label>
+                            <input
+                                type="password"
+                                className="form-control"
+                                name="password"
+                                onChange={this.onChange}
+                                value={password}
+                            />
+                        </div>
+                        <div className="form-group">
+                            <label>Confirm Password</label>
+                            <input
+                                type="password"
+                                className="form-control"
+                                name="password2"
+                                onChange={this.onChange}
```

Comment:
user proper names for functions.

Reply:
this function literally handles the changes occurring in the tag.